src/preamble.js

```diff
@@ -1082,22 +1082,18 @@ function createWasm() {
   // path.
   if (Module['instantiateWasm']) {
 
-#if USE_OFFSET_CONVERTER && PTHREADS
-    if (ENVIRONMENT_IS_PTHREAD) {
+#if USE_OFFSET_CONVERTER
 #if ASSERTIONS
-      assert(Module['wasmOffsetData'], 'wasmOffsetData not found on Module object');
+{{{ runIfWorkerThread("assert(Module['wasmOffsetData'], 'wasmOffsetData not found on Module object');") }}}
```

Comment:
I think this would perhaps be move readable if just the precondition was a macro.

e.g.:
```
if ({{{ runningInWorker() }}}) {
   assert(Module['wasmOffsetData'], 'wasmOffsetData not found on Module object');") }}}
}
```

That way we not writing JS code inside the macro itself.  WDTY?

Reply:
> I think this would perhaps be move readable if just the precondition was a macro.
> 
> e.g.:
> 
> ```
> if ({{{ runningInWorker() }}}) {
>    assert(Module['wasmOffsetData'], 'wasmOffsetData not found on Module object');") }}}
> }
> ```
> 
> That way we not writing JS code inside the macro itself. WDTY?

`runningInWorker` mimics `runningInMain`, as shown below:

https://github.com/emscripten-core/emscripten/blob/e18753e6fbbf2d690e5a750165c693bb6bfec122/src/preamble.js#L735

Reply:
Fair enough.